File: libs/typeorm/src/entities/Timer.ts
import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';

import { Channel } from './Channel.js';

@Entity('channels_timers')
export class Timer {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ManyToOne(() => Channel, c => c.timers)
  @JoinColumn()
  channel: Channel;

  @Column()
  name: string;

  @Column({ default: true })
  enabled: boolean;

  @Column('simple-array', { default: '[]' })
  responses: string[];

  @Column({ default: 0 })
  last: number;

  @Column({ default: 0 })
  timeInterval: number;

  @Column({ default: 0 })
  messageInterval: number;
}

File: apps/api-gql/internal/entity/timer.go
package entity

import (
	"github.com/google/uuid"
	"github.com/twirapp/twir/libs/bus-core/bots"
)

type Timer struct {
	ID                       uuid.UUID
	ChannelID                string
	Name                     string
	Enabled                  bool
	TimeInterval             int
	MessageInterval          int
	LastTriggerMessageNumber int
	Responses                []Response
}

type Response struct {
	ID            uuid.UUID
	Text          string
	IsAnnounce    bool
	Count         int
	AnnounceColor bots.AnnounceColor
}

var TimerNil = Timer{}
